package com.rinko1231.ccb.init;

import net.minecraft.core.registries.Registries;
import net.minecraft.network.chat.Component;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.ItemStack;
import net.neoforged.neoforge.registries.DeferredHolder;
import net.neoforged.neoforge.registries.DeferredRegister;


import static com.rinko1231.ccb.CasterCuriosBonus.MOD_ID;



public class TabInit {
    public static final DeferredRegister<CreativeModeTab> TABS = DeferredRegister.create(Registries.CREATIVE_MODE_TAB, MOD_ID);

    public static final DeferredHolder<CreativeModeTab, CreativeModeTab> CCB_TAB = TABS.register(MOD_ID, () -> CreativeModeTab.builder()
            // Set name of tab to display
            .title(Component.translatable("item_group." + MOD_ID))
            // Set icon of creative tab
            .icon(() -> new ItemStack(ItemReg.MIND_TO_MATTER.get()))
            // Add default items to tab
            .displayItems((params, output) -> {
                ItemReg.ITEMS.getEntries().forEach(it -> output.accept(it.get()));
            })
            .build()
    );
}